rule/unused-param.go

```diff
@@ -87,54 +87,64 @@ type lintUnusedParamRule struct {
 }
 
 func (w lintUnusedParamRule) Visit(node ast.Node) ast.Visitor {
+	var (
+		funcType *ast.FuncType
+		funcBody *ast.BlockStmt
+	)
 	switch n := node.(type) {
+	case *ast.FuncLit:
+		funcType = n.Type
+		funcBody = n.Body
 	case *ast.FuncDecl:
-		params := retrieveNamedParams(n.Type.Params)
-		if len(params) < 1 {
-			return nil // skip, func without parameters
-		}
-
 		if n.Body == nil {
-			return nil // skip, is a function prototype
+			return w // skip, is a function prototype
```

Comment:
Here as we as in some other places it was returning nil. Now it returns `w`. Why is that?

Reply:
`Visit` is invoked in [ast.Walk](https://pkg.go.dev/go/ast#Walk). Previously the function returned `nil` to stop walking the AST after entering into a function definition. Because anonymous functions can be defined inside of any code block, the function now needs to walk through function bodies looking for `*ast.FuncLit` nodes.

Reply:
I think this specific check could be `nil`, as there can't be any `FuncLit`s inside a function prototype... But it felt a bit more obviously correct to just return the visitor in all the cases rather than singling this case out.
```suggestion
			return nil // skip, is a function prototype
```